Source/Core/Core/PowerPC/JitArm64/JitArm64_SystemRegisters.cpp

```diff
@@ -247,38 +247,47 @@ void JitArm64::mfspr(UGeckoInstruction inst)
   case SPR_TL:
   case SPR_TU:
   {
-    // The inline implementation here is inaccurate and out of date as of PR3601
-    FALLBACK_IF(true);  // Fallback to interpreted version.
-
-    /*
     ARM64Reg WA = gpr.GetReg();
     ARM64Reg WB = gpr.GetReg();
     ARM64Reg XA = EncodeRegTo64(WA);
     ARM64Reg XB = EncodeRegTo64(WB);
+    ARM64Reg VC = fpr.GetReg();
+    ARM64Reg VD = fpr.GetReg();
+    ARM64Reg SC = EncodeRegToSingle(VC);
+    ARM64Reg SD = EncodeRegToSingle(VD);
 
     // An inline implementation of CoreTiming::GetFakeTimeBase, since in timer-heavy games the
     // cost of calling out to C for this is actually significant.
-    MOVI2R(XA, (u64)&CoreTiming::g_global_timer);
+    MOVP2R(XA, &CoreTiming::g_last_OC_factor_inverted);
+    m_float_emit.LDR(32, INDEX_UNSIGNED, SC, PPC_REG, PPCSTATE_OFF(downcount));
+    m_float_emit.LDR(32, INDEX_UNSIGNED, SD, XA, 0);
+    m_float_emit.FMUL(SC, SC, SD);
+    m_float_emit.FCVTZS(XB, SC);
+    MOVP2R(XA, &CoreTiming::g_slice_length);
     LDR(INDEX_UNSIGNED, XA, XA, 0);
-    MOVI2R(XB, (u64)&CoreTiming::g_fake_TB_start_ticks);
+    SUB(XA, XA, XB);
+    MOVP2R(XB, &CoreTiming::g_global_timer);
+    LDR(INDEX_UNSIGNED, XB, XB, 0);
+    ADD(XA, XA, XB);
+    MOVP2R(XB, &CoreTiming::g_fake_TB_start_ticks);
     LDR(INDEX_UNSIGNED, XB, XB, 0);
     SUB(XA, XA, XB);
-
     // It might seem convenient to correct the timer for the block position here for even more
-    accurate
+    // accurate
     // timing, but as of currently, this can break games. If we end up reading a time *after* the
-    time
+    // time
     // at which an interrupt was supposed to occur, e.g. because we're 100 cycles into a block with
-    only
+    // only
     // 50 downcount remaining, some games don't function correctly, such as Karaoke Party
-    Revolution,
+    // Revolution,
```

Comment:
It would be nice if you could fix the wrapping here instead of just adding `//` to the short lines (i.e. make the second line "`// accurate timing, but as of currently,` [...]")